internal/app/init/pkg/system/conditions/conditions_test.go

```diff
@@ -4,11 +4,160 @@
 
 package conditions_test
 
-import "testing"
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
 
-func TestEmpty(t *testing.T) {
-	// added for accurate coverage estimation
-	//
-	// please remove it once any unit-test is added
-	// for this package
+	"github.com/stretchr/testify/suite"
+
+	"github.com/talos-systems/talos/internal/app/init/pkg/system/conditions"
+)
+
+type ConditionsSuite struct {
+	suite.Suite
+
+	tempDir string
+}
+
+func (suite *ConditionsSuite) SetupSuite() {
+	var err error
+	suite.tempDir, err = ioutil.TempDir("", "talos")
+	suite.Require().NoError(err)
+
+}
+
+func (suite *ConditionsSuite) TearDownSuite() {
+	suite.Require().NoError(os.RemoveAll(suite.tempDir))
+}
+
+func (suite *ConditionsSuite) createFile(name string) (path string) {
+	path = filepath.Join(suite.tempDir, name)
+	f, err := os.Create(path)
+	suite.Require().NoError(err)
+
+	suite.Require().NoError(f.Close())
+
+	return
+}
+
+func (suite *ConditionsSuite) TestFileExists() {
+	exists, err := conditions.FileExists("no-such-file")(context.Background())
+	suite.Require().NoError(err)
+	suite.Require().False(exists)
+
+	exists, err = conditions.FileExists(suite.createFile("a.txt"))(context.Background())
+	suite.Require().NoError(err)
+	suite.Require().True(exists)
+}
+
+func (suite *ConditionsSuite) TestWaitForFileToExist() {
+	path := suite.createFile("w.txt")
+
+	exists, err := conditions.WaitForFileToExist(path)(context.Background())
+	suite.Require().NoError(err)
+	suite.Require().True(exists)
+
+	suite.Require().NoError(os.Remove(path))
+
+	errCh := make(chan error)
+
+	go func() {
+		exists, err = conditions.WaitForFileToExist(path)(context.Background())
+		suite.Require().True(exists)
+
+		errCh <- err
+	}()
+
+	time.Sleep(50 * time.Millisecond)
+
+	select {
+	case <-errCh:
```

Comment:
wouldn't this fail when `err = nil` ?

Reply:
it doesn't matter at this point what is the error, but condition should be still running at this point (waiting for file to be created), so if condition returns, that is the bug